TESTS/mbedmicro-rtos-mbed/mutex/main.cpp

```diff
@@ -224,46 +219,40 @@ void test_dual_thread_lock(void)
     osStatus stat;
     Thread thread(osPriorityNormal, TEST_STACK_SIZE);
 
-    stat = mutex.lock();
-    TEST_ASSERT_EQUAL(osOK, stat);
+    mutex.lock();
 
     thread.start(callback(F, &mutex));
 
     wait_ms(TEST_LONG_DELAY);
 
-    stat = mutex.unlock();
-    TEST_ASSERT_EQUAL(osOK, stat);
+    mutex.unlock();
 }
 
 /** Test single thread lock recursive
 
     Given a mutex and a single running thread
     When thread calls @a lock twice and @a unlock twice on the mutex
-    Then the returned statuses are osOK
+    Then @a lock and @a unlock operations are successfully performed.
 */
 void test_single_thread_lock_recursive(void)
 {
     Mutex mutex;
     osStatus stat;
 
-    stat = mutex.lock();
-    TEST_ASSERT_EQUAL(osOK, stat);
+    mutex.lock();
 
-    stat = mutex.lock();
-    TEST_ASSERT_EQUAL(osOK, stat);
+    mutex.lock();
```

Comment:
We don;t return any status, but how do we verify if recursive lock here was successful or not?

Reply:
In the original version returned status has been checked. Since we do not have status now we can only execute two successive locks and two successive unlocks and show that the test case has been executed successfully. We do not have status, but please note that ASSERT has been added to the lock/unlock. In case of failure ASSERT will force the test to fail.